### ISSUE TITLE
Assertion `has ("entry")' failed

### ISSUE THREAD

**Issue by samhh** [GothenburgBitFactory/taskwarrior#2018]
#### To report a bug...

* What happened?

I created a series of tasks and projects last night and put my laptop to sleep. Since waking my laptop this morning I receive the the following error when running `task`, `task next`, or `task list` (other commands e.g. add, done, etc appear to work fine):

```
task: /build/task/src/task-2.5.1/src/Task.cpp:1955: float Task::urgency_age() const: Assertion `has ("entry")' failed.
zsh: abort (core dumped)  task
```

* Paste the output of the ```task diag``` command.

```
task 2.5.1
   Platform: Linux

Compiler
    Version: 5.3.0
       Caps: +stdc +stdc_hosted +LP64 +c8 +i32 +l64 +vp64 +time_t64
 Compliance: C++11

Build Features
      Built: Mar  9 2016 20:12:54
      CMake: 3.4.3
    libuuid: libuuid + uuid_unparse_lower
  libgnutls: 3.4.10
 Build type: 

Configuration
       File: /home/sam/.taskrc (found), 1380 bytes, mode 100644
       Data: /home/sam/.task (found), dir, mode 40755
    Locking: Enabled
         GC: Enabled
    $VISUAL: nvim
     Server: 
      Trust: strict
Certificate: , not readable, 0 bytes
        Key: , not readable, 0 bytes
    Ciphers: NORMAL
      Creds: 

Hooks
     System: Enabled
   Location: /home/sam/.task/hooks
             (-none-)

Tests
      $TERM: xterm-kitty (212x51)
       Dups: Scanned 56 tasks for duplicate UUIDs:
             No duplicates found
 Broken ref: Scanned 56 tasks for broken references:
             No broken references found
```

**Comment by pbeckingham** [comment]
It looks like you broke the parser, well done.  Do you have the exact command you entered, to cause this?  We'll need that to fix it.

**Comment by SamHH** [comment]
Biggest accomplishment all week for me :confetti_ball: :laughing: 

I don't unfortunately, but I strongly suspect it was some combination of a URL and surrounding parenthesis if that's of any help.

If it happens again I'll be sure to post the command here!

**Comment by pbeckingham** [comment]
Thanks, that would be appreciated.

**Comment by SamHH** [comment]
Had a thought to look at my zsh histfile! I was converting all of my Todoist tasks over so I don't know which task did it. I don't want to just share it as there's some personal data there. Here's a subset of all the tasks I added with any abnormal formatting in the order in which I used them with the regular text obfuscated, hopefully it helps. (These looked less stupid in terms of formatting in their original forms, I promise haha)

```bash
task add a b \- c\'d e project:x.y
task add a, b project:x due:2018-06-01 wait:2018-06-01
task add 'a (b), c/ (d?))' project:x
task add a! project:x priority:L
task add 'a (b!)' project:x due:wednesday recur:fortnightly
task add 'a (b!)' project:x due:wednesday recur:weekly
task add a \(b\) project:x
task add a b\'c d (e) - f, g project:x priority:M
task add 'a b\'c d (e) - f, g' project:x priority:M
task add "a b\'c d (e) - f, g" project:x priority:M
task add "a b\'c d (e)" project:x
```
I'd also be happy to test these if you can confirm that _all_ the data is in `~/.task/`, in which case I can just back up that directory and restore afterwards.

**Comment by todkrank** [comment]
I annotated a task "Nokia 5.3". To be exact, without the brackets:
**task 29 annotate Nokia 5.3**
Casting **task 29 info** caused the core dump. **task next** didn't work either (because it implies showing annotations, I assume).
Removing the annotation helped.
**task 29 denotate**

Perhaps the full stop symbol "." is somehow responsible, otherwise, no idea what happened. This is the message I got in terminal:
> /usr/include/c++/10/bits/basic_string.h:1019: std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::const_reference std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::operator[](std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type) const [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::const_reference = const char&; std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]: Assertion '__pos <= size()' failed.
> Aborted (core dumped)
>

**Comment by tbabej** [comment]
@todkrank This is also fixed now on 2.6.0 :slightly_smiling_face: 

```
[warrior@tw-2.6.0 ~]$ task add test
Created task 1.
[warrior@tw-2.6.0 ~]$ task 1 annotate Nokia 5.3?
Annotating task 1 'test'.
Annotated 1 task.
Filter: ( ( ( id == 1 ) ) )
[warrior@tw-2.6.0 ~]$ task
[task next]

ID Age  Description             Urg 
 1 9s   test                     0.8
          2021-09-25 Nokia 5.3?

1 task
```

**Comment by tbabej** [comment]
@samhh I was successful in adding all of the above tasks on 2.6.0:

```
ID Age   P Project Recur Due     Description             Urg 
 3  1min   x             -3.3y   task add a, b             13
 7  1min   x         P1D    3d   a (b!)                  8.35
 9  1min   x         P7D    3d   a (b!)                  8.35
11 47s   M x                     a b'c d (e) - f, g       4.9
12 15s   M x                     a b\                     4.9
13  6s   M x                     a b\                     4.9
 5  1min L x                     task a!                  2.8
 2  1min   x.y                   a b - c'd e                1
 4  1min   x                     a (b), c/ (d?))            1
10  1min   x                     a (b)                      1
14  1s     x                     a b\                       1
 1  3min                         test                     0.8
                                   2021-09-25 Nokia 5.3?     
```

Given what commands you have extracted from the history, I believe this is the same issue as #2189, which we fixed a month ago :slightly_smiling_face:
